src/ert/run_models/base_run_model.py

```diff
@@ -539,8 +542,53 @@ async def run_monitor(self, ee_config: EvaluatorServerConfig) -> bool:
 
         return True
 
+    async def run_ensemble_evaluator_async(
+        self, run_context: RunContext, ee_config: EvaluatorServerConfig
+    ) -> List[int]:
+        if not self._end_queue.empty():
+            event_logger.debug("Run model canceled - pre evaluation")
+            self._end_queue.get()
+            return []
+        ensemble = self._build_ensemble(run_context)
+        evaluator = EnsembleEvaluatorAsync(
+            ensemble,
+            ee_config,
+            run_context.iteration,
+        )
+        evaluator_task = asyncio.create_task(
+            evaluator.run_and_get_successful_realizations()
+        )
+        if not (await self.run_monitor(ee_config)):
```

Comment:
Is this only running the monitor, or is it also waiting until experiment is done?

Reply:
hm.... You might be right. Need to rethink this a bit